### PR TITLE
Feat: Show notes panel on crm page

### PR DESCRIPTION
**Pull request by nithindavid**
image

### CODE REVIEW

File: app/javascript/dashboard/api/contactNotes.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import ApiClient from './ApiClient';
+
+class ContactNotes extends ApiClient {
+  constructor() {
+    super('contact_notes', { accountScoped: true });
+  }
+}
+
+export default new ContactNotes();
```
Comment (muhsin-k) [inline app/javascript/dashboard/api/contactNotes.js:9]: Add specs for `contactNotes` API

File: app/javascript/dashboard/modules/contact/components/ManageLayout.vue [hunk 2]
```diff
@@ -67,4 +67,9 @@ export default {
   border-right: 1px solid var(--color-border);
   border-left: 1px solid var(--color-border);
 }
+
+.right {
+  /* background: white; */
```
Comment (muhsin-k) [inline app/javascript/dashboard/modules/contact/components/ManageLayout.vue:72]: Remove commented line

File: app/javascript/dashboard/store/mutation-types.js [hunk 3]
```diff
@@ -154,4 +154,11 @@ export default {
   SET_CAMPAIGNS: 'SET_CAMPAIGNS',
   ADD_CAMPAIGN: 'ADD_CAMPAIGN',
   EDIT_CAMPAIGN: 'EDIT_CAMPAIGN',
+
+  // Canned Response
```
Comment (muhsin-k) [inline app/javascript/dashboard/store/mutation-types.js:158]: Contact notes

**Review by muhsin-k — COMMENTED** [review]
@nithindavid Nice work. Please see the inline comments

File: app/javascript/dashboard/api/contactNotes.js (hunk 1 above)
Comment (nithindavid) [reply app/javascript/dashboard/api/contactNotes.js:9]: The api is not yet ready for contacts. I'll add once API is ready in another PR

File: app/javascript/dashboard/modules/contact/components/ManageLayout.vue [hunk 4]
```diff
@@ -38,13 +39,12 @@ export default {
...
+    this.fetchContactDetails();
   },
   methods: {
-    getContactDetails() {
-      if (this.contactId) {
-        this.$store.dispatch('contacts/show', { id: this.contactId });
-      }
+    fetchContactDetails() {
+      const { contactId } = this;
+      if (contactId) this.$store.dispatch('contacts/show', { id: contactId });
```
Comment (pranavrajs) [inline app/javascript/dashboard/modules/contact/components/ManageLayout.vue:47]: Will there be a case when contactId is nil?

File: app/javascript/dashboard/modules/notes/NotesOnContactPage.vue [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+import NoteList from './components/NoteList';
+
+export default {
+  components: {
+    NoteList,
+  },
+  props: {
+    contactId: {
+      type: [String, Number],
+      default: 0,
```
Comment (pranavrajs) [inline app/javascript/dashboard/modules/notes/NotesOnContactPage.vue:15]: Can't this be a required attribute?

File: app/javascript/dashboard/store/modules/contactNotes.js [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+
+  [types.default.SET_CONTACT_NOTES]($state, { data, contactId }) {
+    Vue.set($state.records, contactId, data);
+  },
+  [types.default.ADD_CONTACT_NOTE]($state, { data, contactId }) {
+    const conatctNotes = $state.records[contactId] || [];
+    $state.records[contactId] = [...conatctNotes, data];
+  },
+  [types.default.DELETE_CONTACT_NOTE]($state, { noteId, contactId }) {
+    const conatctNotes = $state.records[contactId];
```
Comment (pranavrajs) [inline app/javascript/dashboard/store/modules/contactNotes.js:95]: spelling mistake

File: app/javascript/dashboard/store/modules/contactNotes.js [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+      ..._state.uiFlags,
+      ...data,
+    };
+  },
+
+  [types.default.SET_CONTACT_NOTES]($state, { data, contactId }) {
+    Vue.set($state.records, contactId, data);
+  },
+  [types.default.ADD_CONTACT_NOTE]($state, { data, contactId }) {
+    const conatctNotes = $state.records[contactId] || [];
```
Comment (pranavrajs) [inline app/javascript/dashboard/store/modules/contactNotes.js:91]: spelling mistake

**Review by muhsin-k — DISMISSED** [review]

**Comment by muhsin-k** [comment]
@nithindavid The code looks good to me. Please fix the specs

**Pull request merged**